src/platform_impl/ios/monitor.rs

```diff
@@ -1,18 +1,44 @@
 use std::{
-    collections::{HashSet, VecDeque},
+    collections::{BTreeSet, VecDeque},
     fmt,
     ops::{Deref, DerefMut},
 };
 
 use crate::{
     dpi::{PhysicalPosition, PhysicalSize},
-    monitor::VideoMode,
+    monitor::{MonitorHandle as RootMonitorHandle, VideoMode as RootVideoMode},
+    platform_impl::platform::ffi::{id, nil, CGFloat, CGRect, CGSize, NSInteger, NSUInteger},
 };
 
-use crate::platform_impl::platform::ffi::{
-    id, nil, CGFloat, CGRect, CGSize, NSInteger, NSUInteger,
-};
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct VideoMode {
```

Comment:
I don't think it's the right solution to move these fields to the `RootVideoMode`. If we look at how the rest of the rest of the library is architectured, the `RootMonitorHandle` and `RootWindow` only contain a single field, which is the platform-specific implementation. This makes a lot of sense to me, and it'd be a shame to add fields to the "root" types that may or may not be applicable to all platforms. It's boilerplate, yes, but this makes future changes significantly easier in the platform-specific backends.

In fact, we don't really need to the `size`, `bit_depth` and `refresh_rate` fields at all on most platforms, since we already have the native mode handle that contains this information. I'd be inclined to modify the functions to fetch the values from the native mode handle (i.e. `DEVMODEW`, `CGDisplayModeRef`, etc.) instead. I think Wayland is the only one that actually requires us to store the values because of its callback-oriented nature (and incidentally X11, because changing this would require a significant refactor, as we need to store the `XRRScreenResources` instead of re-querying it on every call - although I think this refactor is inevitable anyway as the current `MonitorHandle` returns out of date information if the display configuration changes, but I digress).